domain/pricing.go

```diff
@@ -142,6 +142,15 @@ type PricingWorker interface {
 	RegisterListener(listener PricingUpdateListener)
 }
 
+// PricingUpdateListener defines the interface for the pricing update listener.
 type PricingUpdateListener interface {
-	OnPricingUpdate(ctx context.Context, height int64, pricesBaseQuoteDenomMap map[string]map[string]any, quoteDenom string) error
+	// OnPricingUpdate notifies the listener of the pricing update.
+	OnPricingUpdate(ctx context.Context, height uint64, blockMetaData BlockPoolMetadata, pricesBaseQuoteDenomMap PricesResult, quoteDenom string) error
 }
+
+// PricesResult defines the result of the prices.
+// [base denom][quote denom] => price
+// Note: BREAKING API - this type is API breaking as it is serialized to JSON.
+// from the /tokens/prices endpoint. Be mindful of changing it without
+// separating the API response for backward compatibility.
+type PricesResult map[string]map[string]osmomath.BigDec
```

Comment:
See the map of string to a map of string to BigDed here. It'll be good if we can have an example of the structure here.